src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -115,6 +115,21 @@
       (http/error-response (tru "must accept {0}" content-type)
                            http/status-not-acceptable))))
 
+(defn verify-content-encoding
+  "Verification for the specified list of content-encodings."
+  [app content-encodings]
+  {:pre [(coll? content-encodings)
+         (every? string? content-encodings)]}
+  (fn [{:keys [headers request-method] :as req}]
+    (let [content-encoding (headers "content-encoding")]
+      (if (or (not= request-method :post)
+              (empty? content-encoding)
+              (some #{content-encoding} content-encodings))
```

Comment:
I went with the `some` approach here for consistency with the pre-existing code in the similar [verify-content-type](https://github.com/puppetlabs/puppetdb/blob/1c98f73ee23624049ae9dcf2294bfe9469fc7c61/src/puppetlabs/puppetdb/middleware.clj#L128), but I'm good with a `set` approach you suggested instead if you think that reads better.